presto-tests/src/test/java/com/facebook/presto/memory/TestMemoryManager.java

```diff
@@ -85,35 +88,46 @@ public void shutdown()
         executor = null;
     }
 
-    @Test(timeOut = 240_000)
-    public void testResourceOverCommit()
+    @AfterMethod
+    public void cleanup()
+    {
+        for (TestingPrestoServer server : queryRunner.getServers()) {
+            closeQuietly(server);
+        }
+    }
+
+    @BeforeGroups(groups = {"resourceOverCommit"})
+    public void resourceOverCommit()
             throws Exception
     {
         Map<String, String> properties = ImmutableMap.<String, String>builder()
                 .put("query.max-memory-per-node", "1kB")
                 .put("query.max-total-memory-per-node", "1kB")
                 .put("query.max-memory", "1kB")
                 .build();
+        queryRunner = createQueryRunner(properties);
```

Comment:
there are a couple tests that only set query memory limits as the properties.  Could we do that as session properties within the tests and have those tests share a query runner?  That can help speed up our tests overall